### PR TITLE
add auth page + folder refactor

### PR DESCRIPTION
**Pull request by nayyyhaa**
## In this PR :
-  I've just added auth integration from the mock backend.
- refactored folder structure.
- NOTE: private routes have not been integrated yet.

#### Pages

- [x] [pages/Authorisation.js](https://github.com/nayyyhaa/inNotion-care-react/commit/70f5461ee0fa172bfa9692c6a6cc4624c4572769#diff-61eac70e72073b41844d2b2abd9b7c31160f616e55e5ac0884361ad87c57347a)
- [x] [pages/ProfilePage.js](https://github.com/nayyyhaa/inNotion-care-react/commit/70f5461ee0fa172bfa9692c6a6cc4624c4572769#diff-8f87910fb2d518acef30276ee347304b96916c28c629c92bc779a0422320cdc3)

#### Component files

- [x] [components/LoginForm.js](https://github.com/nayyyhaa/inNotion-care-react/commit/70f5461ee0fa172bfa9692c6a6cc4624c4572769#diff-eb404d8548d8aee44f45218a7d1cbc9bb6619404e5c89b23af03658d6a3e7e03)
- [x] [components/SignupForm.js](https://github.com/nayyyhaa/inNotion-care-react/commit/70f5461ee0fa172bfa9692c6a6cc4624c4572769#diff-50f272adff71bcaff061da9b6e6f43d170bc598e85f462b2de5bfc222b93756b)

 #### Contexts 

- [x] [contexts/AuthContext.js](https://github.com/nayyyhaa/inNotion-care-react/commit/70f5461ee0fa172bfa9692c6a6cc4624c4572769#diff-a8558bdda09a87ea852e53f4861f37246adfd6d1cbbb861d111ac240ccd62363)

#### utils:
- [x] [utils/authService.js](https://github.com/nayyyhaa/inNotion-care-react/commit/70f5461ee0fa172bfa9692c6a6cc4624c4572769#diff-b69b4726d8df6f8c93ca55fb1a865d0331bc9ca9b79160862cc80b867d271060)
 
#### files to *ignore*
- [ ] package.json files
- [ ] files with just path/folder changed
- [ ] CSS files

### Live preview : https://deploy-preview-5--innotion-carev2.netlify.app/

### CODE REVIEW

File: src/contexts/AuthContext.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  const navigate = useNavigate();
+
+  const loginHandler = async (e, email, password) => {
+    e.preventDefault();
+    const [token, userData] = await loginService("/api/auth/login", email, password);
+    setUser(userData);
+    localStorage.setItem("token", token);
+    localStorage.setItem("isAuth", true);
+    localStorage.setItem("user", JSON.stringify(userData));
```
Comment (partha8) [inline src/contexts/AuthContext.js:23]: Hey, why are you saving the userData in the localStorage? How are you going to use it?

Comment (nayyyhaa) [reply src/contexts/AuthContext.js:23]: On refreshing, the user data was getting deleted. I thought of displaying user data like name, email in profile page if user has logged in by saving it in localStorage.

**Review by partha8 — APPROVED** [review]
Good readable code and I was able to understand the flow as well. Nothing new to suggest. Good to merge

File: src/contexts/AuthContext.js (hunk 1 above)
Comment (partha8) [reply src/contexts/AuthContext.js:23]: oh! got it

Comment (partha8) [reply src/contexts/AuthContext.js:23]: What happens when a new user logs in? The old userData will get overwritten right? Also, animesh has implemented a better approach for "remember me" approach, where upon refresh the data won't be deleted.

Comment (nayyyhaa) [reply src/contexts/AuthContext.js:23]: Yes, old data will be deleted from localStorage by logout function.
Oh nice, I'll definitely check that out!

**Review by yash9263 — APPROVED** [review]

**Comment by yash9263** [comment]
Clean and clear written code I found nothing new to suggest.
Nice use of util functions.

**Comment by nayyyhaa** [comment]
Thanks @partha8 & @yash9263 for reviewing. Merging the PR.

**Pull request merged**